### ISSUE TITLE
Continue training with pre trained model

### ISSUE THREAD

**Issue by abdou31** [tensorflow/tensorflow#29932]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution: Windows 10 x64
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):1.13.1
- Python version:3.6
- CUDA/cuDNN version:10/7.4
- GPU model and memory:Nvidia GeForce 840m / 4.00 Go

I cloned a CNN facial landmarks project from Github, this repo can be used to detect facial with 68 landmarks ( 68 for x coordinates and 68 for y coordinates). I have a pre-trained model " frozen_inference_graph.pb"
I used this project to trained my own dataset to detect iris region in the human eye, so I prepared my dataset to be used for training by CNN.
My problem is that the iris annotated with 40 landmarks ( 40 for x coordinates and 40 for y coordinates) and the pre-trained model is trained with 68 landmarks ( number of logits 136 (68+68 )).

![redd](https://user-images.githubusercontent.com/19480228/59718798-3441a380-921b-11e9-91c2-a760102760cc.PNG)

I tried to train my model from scratch, but I didn't get good results:
![iriss](https://user-images.githubusercontent.com/19480228/59718868-5a674380-921b-11e9-8b3c-efb4766c875f.PNG)

I want to continue training with this pre-trained model but I get this error:
> InvalidArgumentError (see above for traceback): Restoring from checkpoint failed. This is most likely due to a mismatch between the current graph and the graph from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:
Assign requires shapes of both tensors to match. lhs shape= [1024,80] rhs shape= [1024,136]`

So it was very clear that my problem is in the number of points ( landmarks ) that are not compatibles ( 80 for iris landmarks and 136 for facial landmarks ) .
I need to know it can be possible to modify the frozen model .pb and what about the transfer learning? and how to use it?

**Comment by jvishnuvardhan** [comment]
This is not Build/Installation or Bug/Performance issue. Please post this kind of support questions at [Stackoverflow](https://stackoverflow.com/questions/tagged/tensorflow). There is a big community to support and learn from your questions. GitHub is mainly for addressing bugs in installation and performance. Thanks!

**Comment by abdou31** [comment]
Thanks, but I really need help because I post this issue on StackOverflow and I didn't get any answer. 
If you have any idea about how to solve this problem, please give me your point of view

**Comment by jvishnuvardhan** [comment]
@abdou31 In transfer learning you could freeze weights for all the layers except last layer, remove last layer and add a  last layer that outputs 80 instead of 136, and train using your data. It should work.

There are many landmark detentions models in GitHub. Please create a standalone code to reproduce the issue and support you accordingly. Thanks!

**Comment by abdou31** [comment]
Thanks.
Actually, I do that and I modified the last layers with units=136 instead of 80 and that causes the problem because the pre-trained model is trained with 80 points.
So I think is not possible to continue training with pre-trained with different last layer.

**Comment by jvishnuvardhan** [comment]
@abdou31 It is possible to train pre-trained model (i.e. called as transfer learning). Please create a GitHub gist and share it. If your data is private, you could you public datasets (MNIST, CIFAR, etc). Thanks!

**Comment by abdou31** [comment]
Can you give me instructions to do that? 
Thanks.

**Comment by jvishnuvardhan** [comment]
@abdou31 You could copy paste your code into [google colab](https://colab.sandbox.google.com/notebooks/welcome.ipynb). Run the colab file and the create a gist (under dropdown  `File`). Thanks!

**Comment by jvishnuvardhan** [comment]
Automatically closing this out since I understand it to be resolved, but please let me know if I'm mistaken.Thanks!

**Comment by abdou31** [comment]
I trained from scratch , 
For now this is not important because I have other issues ( using tflite model on Android device )
